chore: Release 0.22.2 (#1313)
paritytech/jsonrpsee: CHANGELOG.md

[Keep a Changelog]: http://keepachangelog.com/en/1.0.0/

## [v0.22.2] - 2024-03-05

> This is a small patch release that exposes the connection details to server methods without breaking changes.

Reviewer: ```suggestion
This is a small patch release that exposes the connection details in server method implementations without breaking changes.
```

> Currently, users can only retrieve the connection ID from these details.

Reviewer: I think it's better to omit this line because connection ID can be extracted by middleware etc...

```suggestion
```

We plan to extend this functionality in jsonrpsee v1.0, although this will necessitate a breaking change.

### [Added]

> - server: Register raw method with connection ID  ([#1297](https://github.com/paritytech/jsonrpsee/pull/1297))

Reviewer: ```suggestion
- server: Register raw method with connection ID ([#1297](https://github.com/paritytech/jsonrpsee/pull/1297))
```

### [Changed]

> - Update Syn 1.0 -> 2.0  ([#1304](https://github.com/paritytech/jsonrpsee/pull/1304))

Reviewer: ```suggestion
- Update Syn 1.0 -> 2.0 ([#1304](https://github.com/paritytech/jsonrpsee/pull/1304))
```
